/**
* The thread class itself throws the following exceptions according to the standard "std::system_error if 
* the thread could not be started. The exception may represent the error condition 
* std::errc::resource_unavailable_try_again or another implementation-specific error condition."
*/

